src/analysis/useless_initializer.d

```diff
@@ -25,45 +32,128 @@ final class UselessInitializerChecker : BaseAnalyzer
 private:
 
     enum key = "dscanner.useless-initializer";
+
     version(unittest)
+    {
         enum msg = "X";
+    }
     else
+    {
         enum msg = `Variable %s initializer is useless because it does not differ from the default value`;
+    }
 
     static immutable strDefs = [`""`, `""c`, `""w`, `""d`, "``", "``c", "``w", "``d", "q{}"];
     static immutable intDefs = ["0", "0L", "0UL", "0uL", "0U", "0x0", "0b0"];
 
+    HashMap!(string, bool) _structCanBeInit;
+    DynamicArray!(string) _structStack;
+    DynamicArray!(bool) _inStruct;
+    DynamicArray!(bool) _atDisabled;
+    DynamicArray!(bool) _inTest;
+
 public:
 
     ///
     this(string fileName, bool skipTests = false)
     {
         super(fileName, null, skipTests);
+        _inStruct.insert(false);
+        _inTest.insert(false);
+    }
+
+    override void visit(const(Unittest) test)
+    {
+        _inTest.insert(true);
+        test.accept(this);
+        _inTest.removeBack();
+    }
+
+    override void visit(const(StructDeclaration) decl)
+    {
+        if (_inTest.back())
+            return;
+
+        _structStack.insert(decl.name.text);
```

Comment:
Hmm are you sure this would be unique? Wouldn't you need proper mangling?
Consider:

```d
struct F {
	struct G {}
	G g;
}
struct G {
	struct G {} // collision
	G g;
}
```